### ISSUE TITLE
sync_determine_operation: `strength=0.7  # Use a consistent strength for determinism` does not pin a model — the comments overstate the guarantee

### ISSUE THREAD

**Issue by bhargavchintam** [promptdriven/pdd#2344]
## Summary

`sync_determine_operation.py` invokes an LLM to resolve ambiguous sync states, with inline comments asserting the call is deterministic. Neither mechanism it names delivers that: `strength` is an *interpolation over whichever models the current machine has API keys for*, not a model pin, and `temperature=0.0` does not make transformer inference reproducible.

The consequence is that the function deciding **what operation PDD performs next** can reach different conclusions on a developer laptop, a CI runner, and a cloud job from identical repository state.

Inspected at commit `817abe2`. Static inspection only; no PDD commands were run.

## Evidence

**The call site** — `pdd/sync_determine_operation.py:6920-6927`:

```python
        # 4. Invoke LLM with caching for determinism
        response = llm_invoke(
            prompt=formatted_prompt,
            input_json={},
            strength=0.7,  # Use a consistent strength for determinism
            temperature=0.0,  # Use temperature 0 for deterministic output
            verbose=False
        )
```

**Why `strength` is not a pin.** `_select_model_candidates()` (`pdd/llm_invoke.py:3515-3524`) starts by filtering the catalog to models the local environment is credentialed for:

```python
def _select_model_candidates(
    strength: float,
    base_model_name: str,
    model_df: pd.DataFrame,
    manifest_by_model: Optional[Dict[str, Any]] = None,
) -> List[Dict[str, Any]]:
    """Selects and sorts candidate models based on strength and availability."""

    available_df = model_df[model_df['api_key'].notna()].copy()
```

`strength` is then applied as an interpolation over that filtered set — `llm_invoke.py:3681-3694`:

```python
    else: # strength > 0.5
        target_elo = base_elo + (strength - 0.5) * (highest_elo - base_elo)
        ...
        candidates = available_df.sort_values(by='sort_metric').to_dict('records')
```

So `strength=0.7` means "70% of the way from the base model toward the highest-ranked model *that this machine can reach*". The bundled catalog `pdd/data/llm_model.csv` has **260 rows** across AWS Bedrock, Anthropic, OpenAI, Google/Vertex, LM Studio, Ollama and others. Selection is further modified by:

- which `*_API_KEY` variables are set (the `notna()` filter above),
- `PDD_SKIP_LOCAL_MODELS` and `PDD_ALLOW_INTERACTIVE` (`llm_invoke.py:3534-3545`),
- user/project catalog overrides at `~/.pdd/llm_model.csv` and `<project>/.pdd/llm_model.csv`, which the README documents as taking precedence over the packaged catalog.

Two machines with different key sets run this "deterministic" decision on different models.

**Why `temperature=0.0` is not sufficient.** Greedy decoding is deterministic only if the logits are. In production serving they are not: batch size varies with load, and non-associative floating-point reduction in normalization, matmul, and attention kernels produces different logits at different batch sizes, so near-ties flip under `argmax`. See [Thinking Machines, *Defeating Nondeterminism in LLM Inference*](https://thinkingmachines.ai/blog/defeating-nondeterminism/) and their [batch-invariant kernels](https://github.com/thinking-machines-lab/batch_invariant_ops). No seed is passed anywhere in `llm_invoke.py` (zero `grep` matches for `seed`).

**What actually provides the stability.** The first comment on that block is the accurate one — the *cache* is doing the work. LiteLLM response caching is configured at import (`llm_invoke.py:1343-1410`, S3/GCS with SQLite fallback) and enabled per call at `:5770`:

```python
                    litellm_kwargs["caching"] = True
```

That is a reasonable engineering choice. But it means determinism holds for the cache's lifetime and scope, not as a property of the operation — a cache miss on a different runner re-samples the decision.

## Why this matters

The affected function is not peripheral. It decides whether `pdd sync` regenerates, patches, tests, or halts for manual merge, including a confidence gate that routes to `fail_and_request_manual_merge` below 0.75 (`sync_determine_operation.py:6939-6952`). A decision path that varies with ambient environment state means the same commit can take different actions in CI than it took locally.

Note that PDD already states the correct position elsewhere — `docs/prompting_guide.md:1453` says replay "does not guarantee bit-for-bit identical generated code, because the LLM call itself may still be nondeterministic." These comments contradict that.

## Proposed fix

1. **Correct the comments.** They currently assert a guarantee the code does not provide, and a reader auditing this file will take them at face value.
2. **Pin the model for internal decision calls.** Use an explicit model identifier rather than a strength interpolation for control-flow decisions, and fail closed if it is unavailable rather than substituting.
3. **Add a `.pddrc` model lock** for user-facing commands, e.g. `model_lock: "anthropic/claude-x@<catalog-hash>"`, with `pdd sync --require-locked-model` for CI so builds fail rather than silently switching compilers.
4. **Document the determinism boundary** in the README to match what `pdd replay` and the prompting guide already say correctly.

Related: #2342 (infrastructure assessment), #2343 (provenance recording).
